charts/streams-app/templates/deployment.yaml

```diff
@@ -220,3 +226,20 @@ spec:
                 path: log4j2.xml
         {{- end }}
       {{- end }}
+  {{- if and .Values.persistence.enabled .Values.statefulSet }}
+  volumeClaimTemplates:
+    - metadata:
+        name: datadir
+      spec:
+        accessModes: [ "ReadWriteOnce" ]
+        resources:
+          requests:
+            storage: "{{ .Values.persistence.size }}"
+        {{- if .Values.persistence.storageClass }}
+        {{- if (eq "-" .Values.persistence.storageClass) }}
```

Comment:
In which case would the storage class be `-` or why would I set it to `-` specifically?

Reply:
Nevermind, you commented it below